addons/event/models/event_slot.py

```diff
@@ -0,0 +1,189 @@
+    _name = "event.slot"
+    _description = "Event Slot"
+    _order = "is_recurrent DESC, date, start_hour, end_hour, id"
+
+    name = fields.Char("Name", compute="_compute_name", store="True")
+    event_id = fields.Many2one("event.event", "Event", ondelete="cascade")
+    start_hour = fields.Float("Starting Hour", required=True, default=8.0)
+    end_hour = fields.Float("Ending Hour", required=True, default=12.0)
+    start_datetime = fields.Datetime("Start Datetimes", compute="_compute_datetimes")
+    end_datetime = fields.Datetime("End Datetimes", compute="_compute_datetimes")
+    is_recurrent = fields.Boolean(compute="_compute_is_recurrent", store=True)
+    # Punctual
+    recurrent_slot_id = fields.Many2one("event.slot", "Recurrent Slot", store=True, ondelete="cascade")
+    date = fields.Date("Date", compute="_compute_date", store=True, readonly=False)
+    # Recurrent
+    weekdays = fields.Many2many("event.slot.weekday", "Weekday", compute="_compute_weekdays", store=True, readonly=False)
+    generated_slot_ids = fields.One2many("event.slot", "recurrent_slot_id")
+
+    @api.constrains("start_hour", "end_hour")
+    def _check_hours(self):
+        for slot in self:
+            if not (0 <= slot.start_hour <= 23.99 and 0 <= slot.end_hour <= 23.99):
+                raise ValidationError(_("A slot hour must be between 0:00 and 23:59."))
+            elif slot.end_hour <= slot.start_hour:
+                raise ValidationError(_("A slot end hour must be later than its start hour.\n%s", slot.name))
+
+    @api.constrains("date", "start_hour", "end_hour")
+    def _check_non_recurrent(self):
+        for slot in self:
+            if not slot.is_recurrent and not slot.recurrent_slot_id:
+                event_start = slot.event_id.date_begin
+                event_end = slot.event_id.date_end
+                if not (event_start <= slot.start_datetime <= event_end) or not (event_start <= slot.end_datetime <= event_end):
+                    raise ValidationError(_("A slot cannot be scheduled outside of the event time range.\n%s", slot.name))
+
+    @api.constrains("date", "weekdays")
+    def _check_recurrence_type(self):
+        for slot in self:
+            if slot.date and slot.weekdays:
+                raise ValidationError(_("A slot cannot have both a date and week day recurrences."))
+            elif not slot.date and not slot.weekdays:
+                raise ValidationError(_("A slot must have a date or week day recurrence."))
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        slots = super().create(vals_list)
+        for slot in slots:
+            if slot.is_recurrent:
+                slot._update_generated_slots()
+        return slots
+
+    def write(self, vals):
+        res = super().write(vals)
+        if self.is_recurrent and any(field in ["event_id", "start_hour", "end_hour", "weekdays"] for field in vals):
+            self._update_generated_slots()
+        return res
```

Comment:
I remember of an issue with a compute at one point so not sure if it's related but seeing this I got the feeling you try to do the job of a compute